### ISSUE TITLE
Grpc Nuget package 1.14.0 and earlier contain Grpc.Core which references old System.Interactive.Async

### ISSUE THREAD

**Issue by bobuva** [grpc/grpc#16244]
### What version of gRPC and what language are you using?
 1.14.0
 
### What operating system (Linux, Windows, …) and version?
 Windows 10, build 17134.165
 
### What runtime / compiler are you using (e.g. python version or version of gcc)
 Building using .NET Framework 4.6 and 4.7
 
### What did you do?
My 'application' is an agent that is called from injected byte code in another .NET assembly. Those mechanics have always worked for me. I put my agent assembly and its dependency dlls into the Global Assembly Cache (GAC) on Windows because the .NET Common Language Runtime (CLR) loader needs to be able to find it when it is called from whatever application I have instrumented.

When the call is made, I receive an exception: 
```
Could not load file or assembly 'System.Interactive.Async, Version=3.0.1000.0, Culture=neutral, PublicKeyToken=94bc3704cddfc263' or one of its dependencies. The system cannot find the file specified.```
```
The app appears to be built with version `3.0.3000.0` of `System.Interactive.Async`, as that is what shows up in the binaries directory after building my project. However, at runtime, it appears Grpc is looking for that older version, `3.0.1000.0`, shown in the exception message above.

I've seen at least one issue, like this [one](https://github.com/grpc/grpc/issues/8656) and this [one](https://github.com/grpc/grpc/issues/7103) that seem to indicate a newer version should be used to build Grpc.Core. But that is not what is found in the latest Nuget package. 
 
### What did you expect to see?
 No exception at runtime.
 
### What did you see instead?
 Exception as displayed above.

**Comment by jtattermusch** [comment]
@bobuva can you provide a minimal repro?

**Comment by bobuva** [comment]
Ok, I'll try to do that in the next couple of days. I'm a little hurried right now.

**Comment by antony-99** [comment]
The NuGet package System.Interactive.Async 3.1.1 referenced by Grpc.Core contains System.Interactive.Async.dll of different versions for net45 (3.0.1000.0) and net46 (3.0.3000.0).

Provided that there are only net45 and netstandard1.5 versions of Grpc.Core.dll in the package, new projects targeting net46 will reference Grpc.Core.dll for net45 (referencing System.Interactive.Async.dll 3.0.1000.0) and System.Interactive.Async.dll for net46 (3.0.3000.0).

```
    <Reference Include="Grpc.Core, Version=1.0.0.0, Culture=neutral, PublicKeyToken=d754f35622e28bad, processorArchitecture=MSIL">
      <HintPath>packages\Grpc.Core.1.14.2\lib\net45\Grpc.Core.dll</HintPath>
    </Reference>
    <Reference Include="System.Interactive.Async, Version=3.0.3000.0, Culture=neutral, PublicKeyToken=94bc3704cddfc263, processorArchitecture=MSIL">
      <HintPath>packages\System.Interactive.Async.3.1.1\lib\net46\System.Interactive.Async.dll</HintPath>
    </Reference>
```

This does not work without bindingRedirect in app.config (or an assembly resolving "hack" in source code).

```
<?xml version="1.0" encoding="utf-8"?>
<configuration>
    <startup> 
        <supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.6.2" />
    </startup>
  <runtime>
    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
      <dependentAssembly>
        <assemblyIdentity name="System.Interactive.Async" publicKeyToken="94bc3704cddfc263" culture="neutral" />
        <bindingRedirect oldVersion="0.0.0.0-3.0.3000.0" newVersion="3.0.3000.0" />
      </dependentAssembly>
    </assemblyBinding>
  </runtime>
</configuration>
```

Please consider switching Grpc.Core to System.Interactive.Async 3.2.0, whose assembles are of the same version (3.2.0.0) for all supported profiles.

**Comment by kskalski** [comment]
I'm compiling a library project with target framework netstandard2.0 and reference to Grpc.Core Version=1.14.2. When I started using 'async' features the compiled dll started referencing System.Interactive.Async 3.0.3000.0, while at the same time referencing version 3.0.1000.0 through Grpc.Core dependency.

Is there a way to force the project to use specific (3.0.1000.0) version? Unfortunately I need to deal with that at compile time, since this version conflict confuses compilation further up the stream.
I tried to specify explicit reference like
```
    <Reference Include="System.Interactive.Async, Version=3.0.1000.0, Culture=neutral, PublicKeyToken=94bc3704cddfc263">
      <AssemblyName>System.Interactive.Async.dll</AssemblyName>
      <SpecificVersion>True</SpecificVersion>
      <HintPath>..\..\build\System.Interactive.Async.dll</HintPath> <!-- 3.0.1000.0 version -->
    </Reference>
```
but it seems to be ignored, the compiled dll still references both versions (which is even more funny, since it happens also when 3.0.3000.0 version is not in packages/ directory)
